kite-tools-parent/kite-tools/src/main/java/org/kitesdk/tools/TransformTask.java

```diff
@@ -96,63 +96,32 @@ public TransformTask setNumWriters(int numWriters) {
   }
 
   public PipelineResult run() throws IOException {
-    boolean runInParallel = true;
-    if (isLocal(from.getDataset()) || isLocal(to.getDataset())) {
-      runInParallel = false;
-    }
-
     PType<T> toPType = ptype(to);
     MapFn<T, T> validate = new CheckEntityClass<T>(to.getType());
 
-    if (runInParallel) {
```

Comment:
I think we need to explicitly set the mapreduce framework to local if either end is local. Here's what I ended up with:

```java
    if (isLocal(from.getDataset()) || isLocal(to.getDataset())) {
      getConf().set("mapreduce.framework.name", "local");
    }
```